### PR TITLE
Delfi-C3 integration

### PR DESCRIPTION
**Pull request by iiacoban42**
- Included Delfi-C3 in the processing pipeline along with commands for ingesting dummy Delfi-C3 data.
- Mark invalid frames (no satellite match). 
- Lots of refactoring and spring cleaning.
- Unit tests.

### CODE REVIEW

File: src/transmission/processing/XTCEParser.py [hunk 1]
```diff
@@ -94,3 +99,14 @@ class XTCEException(Exception):
...
+class SatParsers:
+
+    def __init__(self):
+        
+        self.parsers = {
+            "delfi_pq": XTCEParser("delfipq/Delfi-PQ.xml", "Radio"),
+            "delfi_next": None,
+            "delfi_c3": XTCEParser("delfic3/Delfi-C3.xml", "TLM"),
+            "da_vinci": None
+        }
```
Comment (iiacoban42) [inline src/transmission/processing/XTCEParser.py:112]: @StefanoSperetta do you have any pointers on how fix this error? I get it when running `python manage.py adddelfic3data` (same concept as with adddelfipqdata). It tries to process the data from `delfi-c3.txt` which came from Satnogs and there is only the TLM stream in the XML. I'm not sure what's off about it. 

```
2022-08-24 01:07:06,659 [process_raw_bucket | ERROR] delfi_c3: frame processing error: An error occurred while calling o1020.processStream.
: org.xtce.toolkit.XTCEDatabaseException: Unable to process binary for desired stream: 'TLM'
        at org.xtce.toolkit.XTCETMStream.processStream(XTCETMStream.java:224)
        at org.xtce.toolkit.XTCETMStream.processStream(XTCETMStream.java:254)
        at jdk.internal.reflect.GeneratedMethodAccessor17.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
        at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
        at py4j.Gateway.invoke(Gateway.java:282)
        at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
        at py4j.commands.CallCommand.execute(CallCommand.java:79)
        at py4j.GatewayConnection.run(GatewayConnection.java:238)
        at java.base/java.lang.Thread.run(Thread.java:829)
Traceback (most recent call last):
  File "/app/transmission/processing/XTCEParser.py", line 36, in processTMFrame
    model = self.stream_.processStream(data)
  File "/usr/local/lib/python3.9/site-packages/py4j/java_gateway.py", line 1321, in __call__
    return_value = get_return_value(
  File "/usr/local/lib/python3.9/site-packages/py4j/protocol.py", line 326, in get_return_value
    raise Py4JJavaError(
py4j.protocol.Py4JJavaError: An error occurred while calling o1020.processStream.
: org.xtce.toolkit.XTCEDatabaseException: Unable to process binary for desired stream: 'TLM'
        at org.xtce.toolkit.XTCETMStream.processStream(XTCETMStream.java:224)
        at org.xtce.toolkit.XTCETMStream.processStream(XTCETMStream.java:254)
        at jdk.internal.reflect.GeneratedMethodAccessor17.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
        at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
        at py4j.Gateway.invoke(Gateway.java:282)
        at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
        at py4j.commands.CallCommand.execute(CallCommand.java:79)
        at py4j.GatewayConnection.run(GatewayConnection.java:238)
        at java.base/java.lang.Thread.run(Thread.java:829)


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/app/transmission/processing/process_raw_bucket.py", line 110, in process_raw_bucket
    store_frame(satellite, row["_time"], row["frame"], row["observer"],  link)
  File "/app/transmission/processing/process_raw_bucket.py", line 32, in store_frame
    telemetry = parser.processTMFrame(bytes.fromhex(frame))
  File "/app/transmission/processing/XTCEParser.py", line 49, in processTMFrame
    raise XTCEException(str(ex)) from ex
transmission.processing.XTCEParser.XTCEException: An error occurred while calling o1020.processStream.
: org.xtce.toolkit.XTCEDatabaseException: Unable to process binary for desired stream: 'TLM'
        at org.xtce.toolkit.XTCETMStream.processStream(XTCETMStream.java:224)
        at org.xtce.toolkit.XTCETMStream.processStream(XTCETMStream.java:254)
        at jdk.internal.reflect.GeneratedMethodAccessor17.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
        at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
        at py4j.Gateway.invoke(Gateway.java:282)
        at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
        at py4j.commands.CallCommand.execute(CallCommand.java:79)
        at py4j.GatewayConnection.run(GatewayConnection.java:238)
        at java.base/java.lang.Thread.run(Thread.java:829)
```

Comment (iiacoban42) [inline src/transmission/processing/XTCEParser.py:112]: @StefanoSperetta do you have any pointers on how to fix this error? I get it when running python `manage.py adddelfic3data` (same concept as with `adddelfipqdata`). It tries to process the data from `delfi-c3.txt` which came from Satnogs and there is only the `TLM` stream in the XML. I'm not sure what's off about it.

```
2022-08-24 01:07:06,659 [process_raw_bucket | ERROR] delfi_c3: frame processing error: An error occurred while calling o1020.processStream.
: org.xtce.toolkit.XTCEDatabaseException: Unable to process binary for desired stream: 'TLM'
        at org.xtce.toolkit.XTCETMStream.processStream(XTCETMStream.java:224)
        at org.xtce.toolkit.XTCETMStream.processStream(XTCETMStream.java:254)
        at jdk.internal.reflect.GeneratedMethodAccessor17.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
        at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
        at py4j.Gateway.invoke(Gateway.java:282)
        at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
        at py4j.commands.CallCommand.execute(CallCommand.java:79)
        at py4j.GatewayConnection.run(GatewayConnection.java:238)
        at java.base/java.lang.Thread.run(Thread.java:829)
Traceback (most recent call last):
  File "/app/transmission/processing/XTCEParser.py", line 36, in processTMFrame
    model = self.stream_.processStream(data)
  File "/usr/local/lib/python3.9/site-packages/py4j/java_gateway.py", line 1321, in __call__
    return_value = get_return_value(
  File "/usr/local/lib/python3.9/site-packages/py4j/protocol.py", line 326, in get_return_value
    raise Py4JJavaError(
py4j.protocol.Py4JJavaError: An error occurred while calling o1020.processStream.
: org.xtce.toolkit.XTCEDatabaseException: Unable to process binary for desired stream: 'TLM'
        at org.xtce.toolkit.XTCETMStream.processStream(XTCETMStream.java:224)
        at org.xtce.toolkit.XTCETMStream.processStream(XTCETMStream.java:254)
        at jdk.internal.reflect.GeneratedMethodAccessor17.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
        at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
        at py4j.Gateway.invoke(Gateway.java:282)
        at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
        at py4j.commands.CallCommand.execute(CallCommand.java:79)
        at py4j.GatewayConnection.run(GatewayConnection.java:238)
        at java.base/java.lang.Thread.run(Thread.java:829)


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/app/transmission/processing/process_raw_bucket.py", line 110, in process_raw_bucket
    store_frame(satellite, row["_time"], row["frame"], row["observer"],  link)
  File "/app/transmission/processing/process_raw_bucket.py", line 32, in store_frame
    telemetry = parser.processTMFrame(bytes.fromhex(frame))
  File "/app/transmission/processing/XTCEParser.py", line 49, in processTMFrame
    raise XTCEException(str(ex)) from ex
transmission.processing.XTCEParser.XTCEException: An error occurred while calling o1020.processStream.
: org.xtce.toolkit.XTCEDatabaseException: Unable to process binary for desired stream: 'TLM'
        at org.xtce.toolkit.XTCETMStream.processStream(XTCETMStream.java:224)
        at org.xtce.toolkit.XTCETMStream.processStream(XTCETMStream.java:254)
        at jdk.internal.reflect.GeneratedMethodAccessor17.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
        at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
        at py4j.Gateway.invoke(Gateway.java:282)
        at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
        at py4j.commands.CallCommand.execute(CallCommand.java:79)
        at py4j.GatewayConnection.run(GatewayConnection.java:238)
        at java.base/java.lang.Thread.run(Thread.java:829)
```

**Comment by StefanoSperetta** [comment]
I cleaned up the logging here a little (removing the stacktrace as all errors come from the same source/place and the message error seems clear enough). I also added the frame causing the problem, just to be easier and faster to interpret:

```2022-08-29 08:10:24,567 [process_raw_bucket | ERROR] delfi_pq: frame processing error: org.xtce.toolkit.XTCEDatabaseException: Unable to process binary for desired stream: 'Radio' (223DCFD3A193605121D868D9854350973782342DBE6E69E6302D41DB)```

A dedicated Grafana dashboard would be needed, but, given the number of changes already in this branch, I would merge it in main and create a new issue to address the Grafana dashboards.

**Comment by StefanoSperetta** [comment]
I would merge in main and address the remaining issue (Grafana dashboard) into a separate issue.

**Review by StefanoSperetta — APPROVED** [review]
I would merge in main and address the remaining issue (Grafana dashboard) into a separate issue.

**Pull request merged**